client/components/contact-details-form-fields/index.jsx

```diff
@@ -0,0 +1,448 @@
+			// rather than set a flag.
+			// Multiple renders triggered by formState via `this.setFormState`
+			// prevent it.
+			this.shouldAutoFocusAddressField = true;
+		}
+	}
+
+	handleSubmitButtonClick = event => {
+		event.preventDefault();
+		this.formStateController.handleSubmit( hasErrors => {
+			this.recordSubmit();
+			if ( hasErrors ) {
+				this.focusFirstError();
+				return;
+			}
+			this.props.onSubmit( this.getMainFieldValues() );
+		} );
+	};
+
+	handleFieldChange = event => {
+		const { name, value } = event.target;
+		const { contactDetails } = this.props;
+
+		if ( name === 'country-code' ) {
+			this.formStateController.handleFieldChange( {
+				name: 'state',
+				value: '',
+				hideError: true,
+			} );
+
+			if ( value && ! contactDetails.phone ) {
+				this.setState( {
+					phoneCountryCode: value,
+				} );
+			}
+			this.focusAddressField();
+		}
+
+		this.formStateController.handleFieldChange( {
+			name,
+			value,
+		} );
+	};
+
+	handlePhoneChange = ( { value, countryCode } ) => {
+		this.formStateController.handleFieldChange( {
+			name: 'phone',
+			value,
+		} );
+
+		this.setState( {
+			phoneCountryCode: countryCode,
+		} );
+	};
+
+	getFieldProps = ( name, needsChildRef = false ) => {
+		const ref = needsChildRef
+			? { inputRef: this.getRefCallback( name ) }
+			: { ref: this.getRefCallback( name ) };
+		const { eventFormName, getIsFieldDisabled } = this.props;
+		const { form } = this.state;
+
+		return {
+			labelClass: 'contact-details-form-fields__label',
+			additionalClasses: 'contact-details-form-fields__field',
+			disabled: getIsFieldDisabled( name ) || formState.isFieldDisabled( form, name ),
+			isError: formState.isFieldInvalid( form, name ),
+			errorMessage: ( formState.getFieldErrorMessages( form, camelCase( name ) ) || [] ).join(
+				'\n'
+			),
+			onChange: this.handleFieldChange,
+			value: formState.getFieldValue( form, name ) || '',
+			name,
+			eventFormName,
+			...ref,
+		};
+	};
+
+	createField = ( name, componentClass, additionalProps, needsChildRef ) => {
+		return (
+			<div className={ `contact-details-form-fields__container ${ kebabCase( name ) }` }>
+				{ createElement(
+					componentClass,
+					Object.assign(
+						{},
+						{ ...this.getFieldProps( name, needsChildRef ) },
+						{ ...additionalProps }
+					)
+				) }
+			</div>
+		);
+	};
+
+	getCountryCode() {
+		const { form } = this.state;
+		return get( form, 'countryCode.value', '' );
+	}
+
+	renderContactDetailsFields() {
+		const { translate, needsFax, hasCountryStates, labelTexts } = this.props;
+		const countryCode = this.getCountryCode();
+		const { phoneCountryCode } = this.state;
+
+		return (
+			<div className="contact-details-form-fields__contact-details">
+				{ this.createField(
+					'organization',
+					HiddenInput,
+					{
+						label: translate( 'Organization' ),
+						text: labelTexts.organization || translate( '+ Add organization name' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **15** times:
`translate( 'Add Organization Name' )` *ES Score: 17*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/checkout-whois-form-consolidation)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).